### PR TITLE
Parsing

### PR DESCRIPTION
**Pull request by NoLongerBreathedIn**
Any objections to this behavior?

### CODE REVIEW

File: parser.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+def parse(f):
+    contents = []
+    while line = f.readline():
```
Comment (Byvire) [inline parser.py]: This is invalid syntax.  Assignment doesn't return a value in Python.

File: parser.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        line = line[:-1]
+        stuff = line.split(maxsplit=2);
+        if stuff[0] == "@algviz":
```
Comment (Byvire) [inline parser.py]: This is an error when line contains no non-whitespace characters.

File: parser.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                                 'contents': parse(f)})
+            elif stuff[1] == "end":
+                break
+            else:
+                contents.append({'command': stuff[1],
+                                 'arguments': stuff[2]})
+        else:
+            contents.append(line)
+    return contents
+
```
Comment (Byvire) [inline parser.py]: Please add at least one test case for `parse`.  Also, please move this (and the test case) into the algviz module, preferably in a subdirectory called either `markdown` or `quine`.

Comment (NoLongerBreathedIn) [reply parser.py]: Will create a test-case directory.

**Comment by NoLongerBreathedIn** [comment]
Nu?

**Comment by Byvire** [comment]
Much better, but you should write test code in python using `unittest`.  If you do, you'll find that `algviz.parser.parse` is not defined, because the submodule `parser` (from `parser.py`) is shadowed by the existing submodule named `parser` (the directory `algviz/parser/`).

**Pull request merged**